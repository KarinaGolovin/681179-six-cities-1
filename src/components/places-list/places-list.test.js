import React from 'react';
import renderer from 'react-test-renderer';
import PlacesList from './places-list.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item';

const mock = [
  {
    title: `Test`,
    type: `Test`,
    price: 0,
    picture: ``,
    link: `#`,
    rating: 0,
    isPremium: true,
    id: `001`,
  },
  {
    title: `Test`,
    type: `Test`,
    price: 0,
    picture: ``,
    link: `#`,
    rating: 0,
    isPremium: true,
    id: `002`,
  },
  {
    title: `Test`,
    type: `Test`,
    price: 0,
    picture: ``,
    link: `#`,
    rating: 0,
    isPremium: true,
    id: `003`,
  },
  {
    title: `Test`,
    type: `Test`,
    price: 0,
    picture: ``,
    link: `#`,
    rating: 0,
    isPremium: true,
    id: `004`,
  },
];

const PlacesListWrapped = withActiveItem(PlacesList);

it(`PlaceCard correctly renders`, () => {
  const component = renderer.create(
      <PlacesListWrapped
        offers={mock}
        onActiveItemChange={() => {}}
      />
  ).toJSON();

  expect(component).toMatchSnapshot();
});
