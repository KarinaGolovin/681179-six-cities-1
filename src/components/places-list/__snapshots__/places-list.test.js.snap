// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard correctly renders 1`] = `
<div
  className="places__list "
>
  <article
    className="place-card "
    id={1}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper "
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src=""
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            0
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="rating place-card__rating"
      >
        <div
          className="rating__stars place-card__stars"
        >
          <span
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Test
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Test
      </p>
    </div>
  </article>
  <article
    className="place-card "
    id={2}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper "
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src=""
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            0
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="rating place-card__rating"
      >
        <div
          className="rating__stars place-card__stars"
        >
          <span
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Test
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Test
      </p>
    </div>
  </article>
  <article
    className="place-card "
    id={3}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper "
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src=""
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            0
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="rating place-card__rating"
      >
        <div
          className="rating__stars place-card__stars"
        >
          <span
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Test
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Test
      </p>
    </div>
  </article>
  <article
    className="place-card "
    id={4}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper "
    >
      <a
        href="/offer/4"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src=""
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            0
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="rating place-card__rating"
      >
        <div
          className="rating__stars place-card__stars"
        >
          <span
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        >
          Test
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Test
      </p>
    </div>
  </article>
</div>
`;
